test/unit/gateways/cyber_source_rest_test.rb

```diff
@@ -233,6 +233,166 @@ def test_url_building
     assert_equal "#{@gateway.class.test_url}/pts/v2/action", @gateway.send(:url, 'action')
   end
 
+  def test_successful_credit_card_purchase_single_request_ignore_avs
+    @gateway.expects(:ssl_post).with do |_host, request_body|
+      json_body = JSON.parse(request_body)
+      assert_equal json_body['processingInformation']['authorizationOptions']['ignoreAvsResult'], 'true'
+      assert_nil json_body['processingInformation']['authorizationOptions']['ignoreCvResult']
+      true
+    end.returns(successful_purchase_response)
+
+    options = @options.merge(ignore_avs: true)
+    assert response = @gateway.purchase(@amount, @credit_card, options)
+    assert_success response
+  end
+
+  def test_successful_credit_card_purchase_single_request_without_ignore_avs
```

Comment:
💬 Comment:

Depending on the answer on how those GSF can end-up being available in the @options instance variable maybe this test should be deleted.

Reply:
pending on the reply of spreedly side to know what to do with this...